frontend/src/components/RideModal/Pages/Driver.tsx

```diff
@@ -1,42 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { ModalPageProps } from '../../Modal/types';
 import styles from '../ridemodal.module.css';
 import { Label, Input, Button } from '../../FormElements/FormElements';
-import { useEmployees } from '../../../context/EmployeesContext';
+import { useReq } from '../../../context/req';
 
 const DriverPage = ({ onBack, onSubmit, formData }: ModalPageProps) => {
   const { register, handleSubmit, formState } = useForm({
-    defaultValues: { driver: formData?.driver ?? '' },
+    defaultValues: {
+      driver: formData?.driver ?? '',
+    },
   });
   const { errors } = formState;
-  const { drivers } = useEmployees();
+  const { withDefaults } = useReq();
+  const [refreshed, setRefreshed] = useState(false);
+
+  const { date, pickupTime: startTime, dropoffTime: endTime } = formData!;
   type DriverOption = { id: string; firstName: string; lastName: string };
-  const driverOptions: DriverOption[] = [
-    { id: 'None', firstName: 'None', lastName: '' },
-  ].concat(drivers);
+  const [availableDrivers, setAvailableDrivers] = useState<DriverOption[]>([]);
+
+  useEffect(() => {
+    if (startTime && endTime && date) {
+      fetch(
+        `/api/drivers/available?date=${date}&startTime=${startTime}&endTime=${endTime}`,
+        withDefaults()
+      )
+        .then((res) => res.json())
+        .then((data) => {
+          setAvailableDrivers(data.data);
+          setRefreshed(true);
+        });
+    }
+  });
+
   return (
     <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
       <div className={styles.inputContainer}>
         <div className={styles.drivers}>
-          {driverOptions.map((d) => (
-            <div className={styles.driver} key={d.id}>
-              <Label
-                htmlFor={d.firstName + d.lastName}
-                className={styles.driverLabel}
-              >
-                {d.firstName}
-              </Label>
-              <Input
-                id={d.firstName + d.lastName}
-                className={styles.driverRadio}
-                name="driver"
-                type="radio"
-                value={d.id}
-                ref={register({ required: true })}
-              />
-            </div>
-          ))}
+          {refreshed ? (
+            availableDrivers.length ? (
```

Comment:
maybe create separate booleans for each condition for readability, something like `loading`, `hasNoDrivers`, and `hasDrivers`? then you can separate out the conditional rendering instead of nesting them

Reply:
I just tried breaking them done (instead of nested) but that seems to be not working because displaying drivers / None should happen only if it is `loaded`. 
I did make new booleans and change their names so that it's more readable now.

Reply:
the booleans can depend on each other, like `loading = ...`, `hasNoDrivers = loading && ...`, `hasDrivers = loading & ...`

Reply:
It seems we just need `loaded` now because there's always a None option.

Reply:
I think I've made the change; but GitHub won't let me re-request review because I must make the requested change. Anyway to fix this?